src/dependencyManagement.test.ts

```diff
@@ -158,50 +159,188 @@ describe("dependencyManagement", () => {
       "installation message"
     );
     expect(compositeDisposable.add).toBeCalledTimes(1);
-    expect(mockConsole.log).toBeCalledTimes(1);
+    expect(mockConsole.log).toBeCalledTimes(2);
     expect(mockConsole.log).toBeCalledWith("claimed lock");
+    expect(mockConsole.log).toBeCalledWith("clearing lock");
     expectLockCleared();
   });
 
-  it("waits for lock to be cleared, and doesn't install, if already locked", async () => {
-    (nova.fs.open as jest.Mock).mockReset().mockImplementationOnce(() => {
-      throw new Error("locked");
+  describe("waits for lock to be cleared", () => {
+    beforeEach(() => {
+      mockFile.read.mockImplementation(() => "PID");
+      (nova.fs.open as jest.Mock).mockImplementation((_, mode) => {
+        if (mode == "x") {
+          throw new Error("locked");
+        } else {
+          return mockFile;
+        }
+      });
+      ProcessMock.mockReset().mockImplementation(() => ({
+        onDidExit: jest.fn((cb) => {
+          cb(0);
+          return { dispose: jest.fn() };
+        }),
+        start: jest.fn(),
+      }));
+      (nova.fs.stat as jest.Mock) = jest.fn(() => ({ mtime: new Date() }));
     });
-    (nova.fs.access as jest.Mock)
-      .mockReset()
-      .mockImplementationOnce(() => true)
-      .mockImplementationOnce(() => true)
-      .mockImplementationOnce(() => false);
 
-    const p = installWrappedDependencies(compositeDisposable, {
-      console: mockConsole,
+    it("doesn't install if already locked and exits cleanly once unlocked", async () => {
+      (nova.fs.access as jest.Mock) = jest.fn(() => true);
+
+      const p = installWrappedDependencies(compositeDisposable, {
+        console: mockConsole,
+      });
+
+      expect(nova.fs.open).toBeCalledTimes(1);
+      // called once to try claiming lock
+      expect(nova.fs.open).toBeCalledWith(
+        "/globalStorage/dependencyManagement/LOCK",
+        "x"
+      );
+      expect(nova.fs.access).not.toBeCalled();
+
+      async function verifyUnlockChecks(n: number) {
+        // clean unlock
+        expect(nova.fs.access).toBeCalledTimes(n);
+        expect(nova.fs.access).toHaveBeenLastCalledWith(
+          "/globalStorage/dependencyManagement/LOCK",
+          F_OK
+        );
+
+        // pid unlock
+        expect(nova.fs.open).toBeCalledTimes(n + 1);
+        expect(nova.fs.open).toHaveBeenLastCalledWith(
+          "/globalStorage/dependencyManagement/LOCK"
+        );
+        expect(mockFile.close).toBeCalledTimes(n);
+        expect(Process).toBeCalledTimes(n);
+        expect(Process).toHaveBeenLastCalledWith("/usr/bin/env", {
+          args: ["ps", "-p", "PID"],
```

Comment:
I'm using `ps -p` instead of `kill -0` because it doesn't exit with 1 on permissions issues.